bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/discovery/UsbGatewayDiscoveryService.java

```diff
@@ -0,0 +1,246 @@
+                        try { // wait for onConnected/onConnectionError or connectTimeout
+                            portCheckLatch.await();
+                        } catch (InterruptedException ie) {
+                            logger.warn("[{}] interrupted: {}", currentScannedPortName, ie.getMessage());
+                        }
+                    } catch (OWNException e) {
+                        logger.debug("[{}] OWNException while trying to connect to a ZigBee USB Gateway: {}",
+                                currentScannedPortName, e.getMessage());
+                        cancelConnectTimeout();
+                        endSerialPortScan();
+                    }
+                }
+                logger.debug("[{}] == finished checking port", currentScannedPortName);
+            }
+        });
+        logger.debug("Finished checking all serial ports");
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        scanning = false;
+        cancelConnectTimeout();
+        endSerialPortScan();
+        portCheckLatch.countDown();
+        super.stopScan();
+        logger.debug("Stopped OpenWebNet ZigBee USB Gateway discovery scan");
+    }
+
+    /**
+     * Cancel the timeout and close connection to proceed to next serial port
+     */
+    private void endSerialPortScan() {
+        USBGateway gw = zbGateway;
+        if (gw != null) {
+            gw.closeConnection();
+            zbGateway = null;
+            logger.debug("[{}] endSerialPortScan()", currentScannedPortName);
+        }
+    }
+
+    private void cancelConnectTimeout() {
+        ScheduledFuture<?> ct = connectTimeout;
+        if (ct != null && !ct.isDone()) {
+            ct.cancel(false);
+            ct = null;
+            logger.debug("[{}] timeout cancelled", currentScannedPortName);
+        }
+    }
+
+    /**
+     * Create and notify a new ZigBee USB Gateway thing has been discovered
+     */
+    private void bridgeDiscovered() {
+        USBGateway gw = zbGateway;
+        if (gw != null) {
+            int gatewayZigBeeId = gw.getZigBeeIdAsDecimal();
+            ThingUID gatewayUID = new ThingUID(OpenWebNetBindingConstants.THING_TYPE_ZB_GATEWAY,
+                    Integer.toString(gatewayZigBeeId));
+            Map<String, Object> gwProperties = new HashMap<>(3);
+            gwProperties.put(OpenWebNetBindingConstants.CONFIG_PROPERTY_SERIAL_PORT, gw.getSerialPortName());
+            gwProperties.put(OpenWebNetBindingConstants.PROPERTY_FIRMWARE_VERSION, gw.getFirmwareVersion());
+            gwProperties.put(OpenWebNetBindingConstants.PROPERTY_ZIGBEEID, String.valueOf(gatewayZigBeeId));
+
+            DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(gatewayUID).withProperties(gwProperties)
+                    .withLabel(OpenWebNetBindingConstants.THING_LABEL_ZB_GATEWAY + " (" + gw.getFirmwareVersion() + ", "
+                            + gw.getSerialPortName() + ")")
```

Comment:
Labels are expected to be as short as possible. Guideline is 2-3 words with max 20-25 chars. See https://www.openhab.org/docs/developer/bindings/thing-xml.html#formatting-labels-and-descriptions

Reply:
ok, will remove the fw version, but i think serial port is useful here